packages/venia-ui/targets/intercept.js

```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const loader = require.resolve('./rendererCollectionLoader');
+
+const isRCR = mod =>
+    mod.resource ===
+    path.resolve(
+        __dirname,
+        '../lib/components/RichContent/richContentRenderers.js'
+    );
+
+const loaderIdent = __filename;
+
+const loaderIsInstalled = mod => mod.loaders.some(l => l.ident === loaderIdent);
+
+const name = '@magento/venia-ui';
+
+module.exports = api => {
+    const richContentRenderers = api.getTarget('richContentRenderers');
+    api.getTarget('@magento/pwa-buildpack', 'webpackCompiler').tap(
```

Comment:
I imagine for other parts of the code that declare their own targets in the future this implementation of handling the target will look very similar and won't change that much. Any chance this can be generalised so it can be more easily reused?

Reply:
There's more than a chance; there's a plan! We plan to leave these code snippets in place for a while, as we implement more and more functionality as extensions. As patterns appear, we will abstract them into a library of extension utility functions. This way, we can grow towards a more terse, declarative model, while still supporting lower-level customizations.